decred/tests/unit/dcr/test_txscript.py

```diff
@@ -2352,30 +2352,53 @@ def test_scriptNumBytes(self):
             )
 
 
-class TestVSP(unittest.TestCase):
-    def test_blob(self):
-        pi = vsp.PurchaseInfo(
-            addr="someaddr",
-            fees=1.3,
-            script=ByteArray("0a1b2c"),
-            ticketAddr="someotheraddr",
-            vBits=2,
-            vBitsVer=1,
-            stamp=123456,
-        )
-        stakePool = vsp.VotingServiceProvider(
-            "http://localhost:54321", "somerandomkey", mainnet.Name, pi
-        )
-        b = stakePool.serialize()
-        rePool = vsp.VotingServiceProvider.unblob(b.b)
-        self.assertEqual(stakePool.url, rePool.url)
-        self.assertEqual(stakePool.apiKey, rePool.apiKey)
-        self.assertEqual(stakePool.net.Name, rePool.net.Name)
-        rePI = rePool.purchaseInfo
-        self.assertEqual(pi.poolAddress, rePI.poolAddress)
-        self.assertEqual(pi.poolFees, rePI.poolFees)
-        self.assertEqual(pi.script, rePI.script)
-        self.assertEqual(pi.ticketAddress, rePI.ticketAddress)
-        self.assertEqual(pi.voteBits, rePI.voteBits)
-        self.assertEqual(pi.voteBitsVersion, rePI.voteBitsVersion)
-        self.assertEqual(pi.unixTimestamp, rePI.unixTimestamp)
+def test_is_unspendable():
+    class test:
+        def __init__(
+            self, name, amount, pkScript, want,
+        ):
+            """
+            Args:
+                name (str): Short description of the test.
+                amount (int): Value of the txOut this script spends.
+                pkScript (ByteArray): Spending script.
+                want (bool): Whether the tx is unspendable
+            """
+            self.name = name
+            self.amount = amount
+            self.pkScript = pkScript
+            self.want = want
+
+    # fmt: off
+    tests = [
+        test(
+            name="not spendable: begins with OP_RETURN",
+            amount=100,
+            pkScript=ByteArray([0x6A, 0x04, 0x74, 0x65, 0x73, 0x74]),
+            want=True,
+        ),
+        test(
+            name="not spendable: zero amount",
+            amount=0,
+            pkScript=ByteArray([0x76, 0xa9, 0x14, 0x29, 0x95, 0xa0,
+                               0xfe, 0x68, 0x43, 0xfa, 0x9b, 0x95, 0x45,
+                               0x97, 0xf0, 0xdc, 0xa7, 0xa4, 0x4d, 0xf6,
+                               0xfa, 0x0b, 0x5c, 0x88, 0xac]),
+            want=True,
+        ),
+        test(
+            name="spendable",
+            amount=100,
+            pkScript=ByteArray([0x76, 0xa9, 0x14, 0x29, 0x95, 0xa0,
+                               0xfe, 0x68, 0x43, 0xfa, 0x9b, 0x95, 0x45,
+                               0x97, 0xf0, 0xdc, 0xa7, 0xa4, 0x4d, 0xf6,
+                               0xfa, 0x0b, 0x5c, 0x88, 0xac]),
+            want=False,
+        ),
+    ]
+    # fmt: on
+    for test in tests:
+        res = txscript.isUnspendable(test.amount, test.pkScript)
+        assert (
+            res == test.want
+        ), f"wanted {test.want} for test {test.name} but got {res}"
```

Comment:
I'd keep `wanted` and `got` values close together, and leave the `test.name` at the end.